packages/front-end/components/Auth/Welcome.tsx

```diff
@@ -26,6 +27,13 @@ export default function Welcome({
   const [loading, setLoading] = useState(false);
   const [error, setError] = useState(null);
   const [welcomeMsgIndex] = useState(Math.floor(Math.random() * 4));
+  const { pathname } = useRouter();
+
+  useEffect(() => {
+    if (pathname === "/invitation") {
+      setState("register");
+    }
+  }, []);
```

Comment:
Should `pathname` be a dependency for this useEffect hook?  I'm wondering if there's ever a race condition where `pathname` is not defined on the initial render.

Reply:
Addressed!